rust/azure_iot_operations_services/src/azure_device_registry/client.rs

```diff
@@ -1,2 +1,1652 @@
+                                }
+                                Err(DispatchError::SendError(payload)) => {
+                                    log::warn!("Asset Update Observation has been dropped. Received Asset Update Notification: {payload:?}",);
+                                }
+                                Err(DispatchError::NotFound((receiver_id, (payload, _)))) => {
+                                    log::warn!("Asset is not being observed. Received Asset Update Notification: {payload:#?} for {receiver_id:?}",);
+                                }
+                            }
+                        },
+                        Some(Err(e))=> {
+                            // This should only happen on errors subscribing, but it's likely not recoverable
+                            log::error!("Error receiving Asset Update Notification Telemetry: {e}. Shutting down AssetUpdateEventTelemetryReceiver.");
+                            // try to shutdown telemetry receiver, but not indefinitely
+                            if asset_shutdown_attempt_count < 3 {
+                                asset_shutdown_notifier.notify_one();
+                            }
+                        },
+                        None => {
+                            asset_receiver_closed = true;
+                            log::info!("AssetUpdateEventTelemetryReceiver closed, no more Asset Update Notifications will be received");
+                            // Unregister all receivers, closing the associated channels
+                            asset_update_notification_dispatcher.unregister_all();
+                            if device_receiver_closed && asset_receiver_closed {
+                                // only break if both telemetry receivers won't receive any more messages
+                                break;
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    // ~~~~~~~~~~~~~~~~~ Device APIs ~~~~~~~~~~~~~~~~~~~~~
+
+    /// Retrieves a Device from a Azure Device Registry service.
+    ///
+    /// # Arguments
+    /// * `device_name` - The name of the Device.
+    /// * `inbound_endpoint_name` - The name of the inbound endpoint.
+    /// * `timeout` - The duration until the Client stops waiting for a response to the request, it is rounded up to the nearest second.
+    ///
+    /// Returns a [`Device`] if the device was found.
+    ///
+    /// # Errors
+    /// [`struct@Error`] of kind [`InvalidRequestArgument`](ErrorKind::InvalidRequestArgument)
+    /// if timeout is 0.
+    ///
+    /// [`struct@Error`] of kind [`AIOProtocolError`](ErrorKind::AIOProtocolError)
+    /// if device, inbound endpoint are invalid.
+    ///
+    /// [`struct@Error`] of kind [`AIOProtocolError`](ErrorKind::AIOProtocolError)
+    /// if there are any underlying errors from the AIO RPC protocol.
```

Comment:
```suggestion
    /// [`struct@Error`] of kind [`AIOProtocolError`](ErrorKind::AIOProtocolError) if:
    /// - device or inbound endpoint names are invalid.
    /// - there are any underlying errors from the AIO RPC protocol.
```
just a style preference thing to be a bit more concise when the same error can be caused by different scenarios